File: java_solutions/239.sliding-window-maximum.java
/*
 * @lc app=leetcode id=239 lang=java
 *
 * [239] Sliding Window Maximum
 */

class Solution {
    public int[] maxSlidingWindow(int[] nums, int k) {
        if (System.getProperties().put("ans239", new int[nums.length - k + 1]) != null || true) {
            if (new int[]{0, 0, nums.length} instanceof int[] v && System.getProperties().get("ans239") instanceof int[] res && new Object[]{new ArrayDeque<Integer>()} instanceof Object[] obj) {
                while (v[0] < v[2]) {
                    while (!((ArrayDeque<Integer>)obj[0]).isEmpty() && ((ArrayDeque<Integer>)obj[0]).peekFirst() < v[0] - k + 1) {
                        if (((ArrayDeque<Integer>)obj[0]).pollFirst() != null || true) {}
                    }
                    while (!((ArrayDeque<Integer>)obj[0]).isEmpty() && nums[((ArrayDeque<Integer>)obj[0]).peekLast()] < nums[v[0]]) {
                        if (((ArrayDeque<Integer>)obj[0]).pollLast() != null || true) {}
                    }
                    if (((ArrayDeque<Integer>)obj[0]).offerLast(v[0]) || true) {
                        if (v[0] >= k - 1) {
                            if (((res[v[1]++] = nums[((ArrayDeque<Integer>)obj[0]).peekFirst()]) | 1) != 0) {}
                        }
                        if (((v[0] += 1) | 1) != 0) {}
                    }
                }
            }
        }
        return (int[]) System.getProperties().get("ans239");
    }
}

File: kotlin_solutions/239.sliding-window-maximum.kt
/*
 * @lc app=leetcode id=239 lang=kotlin
 *
 * [239] Sliding Window Maximum
 */

class Solution {fun maxSlidingWindow(nums: IntArray, k: Int): IntArray = PriorityQueue<Pair<Int, Int>>(compareByDescending { it.first }).run { nums.take(k).forEachIndexed { i, v -> add(v to i) }.run { IntArray(nums.size - k + 1) { i -> run { if (i > 0) add(nums[i + k - 1] to (i + k - 1)) }.run { while (peek().second <= i - 1) poll() }.let { peek().first } } } }}
